### PR TITLE
fix: create device_profile workflow before loading workflows from disk

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Run the device profile manager builder before the loading the workflows, otherwise the newly created device_profile workflow will not be detected until the service is restarted.

The problem was being masked in the existing system test as the test restarted the tedge-agent in the setup due to adding an additional workflow (firmware_update). The system test to check the supported operation has been moved to an independent test which does not restart the tedge-agent in the test setup.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

Partially solves https://github.com/thin-edge/thin-edge.io/issues/3088

**Note** installing thin-edge.io via the install.sh script will still require the tedge-agent to be restarted, however this is less of an issue

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by albinsuresh — APPROVED** [review]
LGTM

File: tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_initialization.robot [hunk 1]
```diff
@@ -0,0 +1,24 @@
+*** Settings ***
+Resource            ../../../resources/common.resource
+Library             ThinEdgeIO
+Library             Cumulocity
+Library             OperatingSystem
+Library             Collections
```
Comment (gligorisaev) [inline tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_initialization.robot:6]: are the Libraries OperatinSystem and Collections needed?

**Review by didier-wenzek — APPROVED** [review]
Approved

File: tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_initialization.robot (hunk 1 above)
Comment (albinsuresh) [reply tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_initialization.robot]: Removed the unnecessary libraries.

Comment (reubenmiller) [reply tests/RobotFramework/tests/tedge_agent/device_profile/device_profile_initialization.robot]: Yeah copy paste stuff

**Pull request merged**